src/integrator_interface.jl

```diff
@@ -1,67 +1,102 @@
+    end
 
-    numiters = 1
-    while true
-      if typeof(integrator.u) <: AbstractArray
-        copy!(integrator.u_cache,integrator.u)
-      else
-        integrator.u_cache = integrator.u
-      end
-      perform_step!(integrator,integrator.cache)
+    if eltype(integrator.u) <: Real # not possible to use nlsolve with units (yet)
+      integrator.first_iteration = true # first iteration step needs special updates
 
-      if typeof(integrator.resid) <: AbstractArray
-        integrator.resid .= (integrator.u .- integrator.u_cache)./(integrator.picardabstol .+ max.(abs.(integrator.u),abs.(integrator.u_cache)).*integrator.picardreltol)
+      # execute Anderson acceleration of fixed-point iteration
+      if typeof(integrator.u) <: Vector
+        nlsolve(integrator.iterator,integrator.u;
+	        method=:anderson,m=integrator.m,iterations=integrator.max_fixedpoint_iters,xtol=0,ftol=1)
       else
-        integrator.resid = (integrator.u .- integrator.u_cache)./(integrator.picardabstol .+ max.(abs.(integrator.u),abs.(integrator.u_cache)).*integrator.picardreltol)
-      end
-
-      picardEEst = integrator.picardnorm(integrator.resid)
-      if picardEEst < 1 || numiters > integrator.max_picard_iters
-        break
+        nlsolve(integrator.iterator,vec(collect(integrator.u));
+	        method=:anderson,m=integrator.m,iterations=integrator.max_fixedpoint_iters,xtol=0,ftol=1)
       end
-      if numiters == 1
-        integrator.integrator.tprev = integrator.t
-        integrator.integrator.t = integrator.t+integrator.dt
-        integrator.integrator.uprev = integrator.u
+    else # simple Picard iteration for units
+      for i in 1:integrator.max_fixedpoint_iters
+        # update cache of u
+        if typeof(integrator.u) <: AbstractArray
+          copy!(integrator.u_cache,integrator.u)
+        else
+          integrator.u_cache = integrator.u
+        end
+
+        perform_step!(integrator,integrator.cache)
+
+	# calculate residuals
+        if typeof(integrator.resid) <: AbstractArray
+          @. integrator.resid = (integrator.u - integrator.u_cache)/
+            @muladd(integrator.fixedpoint_abstol + max(abs(integrator.u),abs(integrator.u_cache))*integrator.fixedpoint_reltol)
+        else
+          integrator.resid = @. (integrator.u - integrator.u_cache)/
+            (integrator.fixedpoint_abstol + max(abs(integrator.u),abs(integrator.u_cache))*integrator.fixedpoint_reltol)
+        end
+
+	# special updates of ODE integrator in first iteration step
+        if i == 1
+          integrator.integrator.tprev = integrator.t
+          integrator.integrator.t = integrator.t+integrator.dt
+          integrator.integrator.uprev = integrator.u
+        end
+
+	# update ODE integrator with values of DDE integrator
+        integrator.integrator.u = integrator.u
+        integrator.integrator.k = integrator.k
+
+	# stop fixed-point iteration when residuals are small
+	integrator.picardnorm(integrator.resid) < 1 && break
```

Comment:
Currently the exit condition of the loop is checked directly after calculating the residuals. However, then no updates would be applied to the ODE integrator if the loop is left during the first iteration? For now, I moved it to the end - the same way Anderson accelerated fixed-point iteration with NLsolve does it right now.